### ISSUE TITLE
Invalid parameter "content" not recognized

### ISSUE THREAD

**Issue by kerrykimbrough** [swagger-api/swagger-parser#1646]
Here's how the OpenAPI spec defines the "content" field for a Parameter Object:

> A map containing the representations for the parameter. The key is the media type and the value describes it. The map MUST only contain one entry.

Accordingly, an empty "content" that defines no media types is an error. Likewise a "content" that defines multiple media types. But such invalid parameters are accepted and no error is reported.

Here's how to reproduce this problem:

```java

    @Test
    public void testParamContent() {
        String json =
            "{"
            + "  \"openapi\": \"3.0.0\","
            + "  \"info\": {"
            + "    \"title\": \"Operations\","
            + "    \"version\": \"0.0.0\""
            + "  },"
            + "  \"paths\": {"
            + "    \"/operations\": {"
            + "      \"post\": {"
            + "        \"parameters\": ["
            + "          {"
            + "            \"name\": \"param0\","
            + "            \"in\": \"query\","
            + "            \"content\": {"
            + "            }"
            + "          },"
            + "          {"
            + "            \"name\": \"param1\","
            + "            \"in\": \"query\","
            + "            \"content\": {"
            + "              \"text/plain\": {"
            + "              }"
            + "            }"
            + "          },"
            + "          {"
            + "            \"name\": \"param2\","
            + "            \"in\": \"query\","
            + "            \"content\": {"
            + "              \"text/plain\": {"
            + "              },"
            + "              \"application/json\": {"
            + "                \"schema\": {"
            + "                  \"type\": \"object\""
            + "                }"
            + "              }"
            + "            }"
            + "          }"
            + "        ],"
            + "        \"responses\": {"
            + "          \"default\": {"
            + "            \"description\": \"None\""
            + "          }"
            + "        }"
            + "      }"
            + "    }"
            + "  }"
            + "}"
            ;
        OpenAPIV3Parser parser = new OpenAPIV3Parser();
        SwaggerParseResult result = parser.readContents(json, null, null);
        Operation post = result.getOpenAPI().getPaths().get( "/operations").getPost();

        Parameter param0 =
            post.getParameters().stream()
            .filter( p -> "param0".equals( p.getName()))
            .findFirst()
            .orElseThrow( () -> new IllegalStateException( "Can't find parameter=param0"));
        assertEquals( param0.getContent().size(), 0, "Empty content size"); // No! Should report an error

        Parameter param1 =
            post.getParameters().stream()
            .filter( p -> "param1".equals( p.getName()))
            .findFirst()
            .orElseThrow( () -> new IllegalStateException( "Can't find parameter=param1"));
        assertEquals( param1.getContent().size(), 1, "Valid content size");

        Parameter param2 =
            post.getParameters().stream()
            .filter( p -> "param2".equals( p.getName()))
            .findFirst()
            .orElseThrow( () -> new IllegalStateException( "Can't find parameter=param2"));
        assertEquals( param2.getContent().size(), 2, "Invalid content size"); // No! Should report an error
    }
```

**Comment by kerrykimbrough** [comment]
The OpenAPI spec also has this to say about the Parameter Object:

> A parameter MUST contain either a schema property, or a content property, but not both. 

Accordingly, a parameter with neither schema nor content is invalid, as is a parameter with both. But such invalid parameters are accepted and no error is reported.

**Comment by gracekarina** [comment]
Closed by merged PR #1647 
Thanks @kerrykimbrough
